docs/src/proposals/leader-qos.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Leader QoS
+---
+
+## Motivation:
+Banking thread has too many packets in its buffer, it needs an ability to pack high-priority transactions into limited block space.
+
+## Scope:
+Leader has multiple banking threads, each thread does:
+1. Receives packet_batches from verified_receiver;
+2. Push packet_batches into FIFO buffer;
+3. Traverse FIFO buffer linearly to select packets for processing (aka consuming) or forwarding;
+   Where consuming is a process of preparing selected packets into batches of sanitized_transactions, submitting for attempted bank execution, which in turn includes block cost tracking, account locking, and executing and committing. Retryable transactions will be put back to buffer for next iteration.
+   Where forwarding is a process of select unprocessed packets from unforwarded packet_batches in FIFO order, batch-send to Fwd ports. 
+
+The aim of this proposal is to replace linear FIFO buffer traversing with transaction-based prioritization. The change scope is mainly at step #3, instead of linearly selecting unprocessed packets from buffered packet_batches, it will sort unprocessed packets with following criteria to prioritize consuming or forwarding. 
+
+The consuming and forwarding processing as described above largely remains the same, as they are outside scope of this proposal. 
+
+## Prioritization criteria:
+1. Transaction's fee-per-CU ( `= (additional_fee + base_fee)/requested_CU)` ), higher paid transactions have priority to be first considered for block inclusion or forwarding. 
+2. When fee-per-CU is same, Transactions from higher staked validators have higher chance to be considered for blocking inclusion or forwarding.
```

Comment:
was this in your original pr? thought it was stake weight first but maybe I misread.
this makes sense to me as opposed to pure stake weighted priority

Reply:
It's part of original PR, probably I haven't fully committed the changes when you looked at it. 
Yeah, the prioritizing by fee/CU is better.